core/src/apps/common/cbor.py

```diff
@@ -217,6 +237,19 @@ def __eq__(self, other: object) -> bool:
         )
 
 
+class Set:
+    def __init__(self, array: List[Value]) -> None:
+        self.array = array
```

Comment:
The tuple is not hashable if it contains some mutable stuff (and such stuff is always suspicious and likely not intentional), and it would be nice to learn about it immediately, not only when trying to encode something complicated.

Reply:
I agree with the duplicity check. A Set instance with duplicates isn't really a valid instance. However, when encoding, I simply ignore duplicates - I don't raise an error (because that's how it is done with maps). Should an error be raised? As I am thinking about it now, it makes more sense to raise an error, than to fail silently with incorrect data and maps are quite different, since they can't contain duplicates by definition. What do you think?

**To the second comment:**
But our sets can contain mutable stuff, can't they? 

`Set([[1, 2], [3, 4]])`

The first element is a list i.e mutable. I can create a tuple from it => `([1, 2], [3, 4])` but that is still not hashable. However, it is a valid case for transaction inputs.